props/saber_fett263_buttons.h

```diff
@@ -293,248 +660,3355 @@ class SaberFett263Buttons : public virtual PropBase {
+        char playtrack[128];
+        RunCommandAndGetSingleLine("list_current_tracks", nullptr, track_num_, playtrack, sizeof(playtrack));
+  #ifdef ENABLE_AUDIO
+        MountSDCard();
+        EnableAmplifier();
+        track_player_ = GetFreeWavPlayer();
+        if (track_player_) {
+          track_player_->Play(playtrack);
+        } else {
+          STDOUT.println("No available WAV players.");
+        }
+  #else
+        STDOUT.println("Audio disabled.");
+  #endif
+        break;
+  #ifdef FETT263_EDIT_MODE_MENU
+      case MENU_TOP:
+        menu_top_pos_ += direction;
+        if (menu_top_pos_ <= 0) menu_top_pos_ = SUBMENUS;
+        if (menu_top_pos_ > SUBMENUS) menu_top_pos_ = 1;
+        switch (menu_top_pos_) {
+          case 1:
+            PlayMenuSound("mstyle.wav");
+            break;
+          case 2:
+            PlayMenuSound("mcolor.wav");
+            break;
+          case 3:
+            PlayMenuSound("mfont.wav");
+            break;
+          case 4:
+            PlayMenuSound("mtrack.wav");
+            break;
+          case 5:
+            PlayMenuSound("msetting.wav");
+            break;
+          case 6:
+            PlayMenuSound("mcopy.wav");
+            break;
+          case 7:
+            PlayMenuSound("mdelete.wav");
+            break;
+          case 8:
+            PlayMenuSound("mexit.wav");
+            break;
+       }
+       break;
+  #if NUM_BLADES > 1
+      case MENU_BLADE_STYLE:
+      case MENU_BLADE_COLOR:
+      case MENU_BLADE_SETTING:
+      case MENU_BLADE_COPY:            
+      case MENU_BLADE_LENGTH:
+        blade_num_ += direction;
+        if (blade_num_ > NUM_BLADES) blade_num_ = 1;
+        if (blade_num_ < 1) blade_num_ = NUM_BLADES;
+        BladePreview(blade_num_);
+        SFX_mnum.Select(blade_num_ - 1); 
+        wav_player->PlayOnce(&SFX_mnum, 0.0);
+        break;
+      case MENU_COPY_COLOR:
+      case MENU_RESET_COLOR:           
+      case MENU_COPY:
+      case MENU_DELETE:
+        if (direction > 0) {
+          choice_ = true;
+          PlayMenuSound("mconfirm.wav");
+        } else {
+          choice_ = false;
+          PlayMenuSound("mcancel.wav");
+        }
+        break;
+  #endif
+      case MENU_STYLE:
+        style_num_ += direction;
+        if (style_num_ < 0) style_num_ = num_presets_ - 1;
+        if (style_num_ > num_presets_ - 1) style_num_ = 0;
+        char style_arg[10];
+        itoa(style_num_, style_arg, 10);
+        current_preset_.SetStyle(blade_num_, style_parser.SetArgument(current_preset_.GetStyle(blade_num_), 1, style_arg)); 
+        current_preset_.Save();
+        if (direction > 0) {
+          PlayMenuSound("mup.wav");
+        } else {
+          PlayMenuSound("mdown.wav");
+        }
+        UpdateStyle(current_preset_.preset_num);
+        break;
+      case MENU_COLOR:
+        break;
+      case MENU_COLOR_SUB:
+  #if NUM_BLADES == 1
+        menu_sub_pos_ += direction;
```

Comment:
move common parts before/after #if/#else/#endif

Reply:
updated

Reply:
still more common parts to move. (maybe you forgot to upload something?)

Reply:
Noted in other location, the number of options and case are different, I'm not seeing any "common" part that could easily be separated unless I'm missing something.